snapshot-labs/snapshot
# fix: fix check for proposal passing

### Issues
theres a bug where someone could request osnap execution after the vote ends, but before it finalizes. its possible during the finalization step that vote tallys may change. not only that but submission before finalization will be disputed by our automated bot infrastructure, leading to loss of bonds. 

Fixes #

### Changes 
*_(Briefly describe the changes made in this PR)_*

1. this adds a check to ensure the proposal.score_state is final

### How to test
*_(Explain how the changes can be tested, including any required setup steps)_*

1. in a test space with osnap enabled, run through a completet vote, and see that you cannot request execution before the vote is finalized.


### Self-review checklist

### Additional notes or considerations
*_(Include any other relevant information or context that may be helpful for the reviewer)_*
